src/main.rs

```diff
@@ -0,0 +1,482 @@
+    running: bool,
+    cursor_pos: (u16, u16),
+    mouse_pressed: bool,
+    sample_rate: f32,
+    apply_window: bool,
+    target_frequency: Option<f32>,
+    graph: Graph,
+}
+
+impl Default for State {
+    fn default() -> Self {
+        State {
+            running: true,
+            apply_window: true,
+            target_frequency: None,
+            graph: Graph::AmplitudeSpectrum,
+            cursor_pos: (0, 0),
+            sample_rate: 0.0,
+            mouse_pressed: false,
+        }
+    }
+}
+
+#[rustfmt::skip]
+mod ascii_symbols;
+mod algo;
+
+fn draw_graph<B>(
+    frame: &mut Frame<B>,
+    area: Rect,
+    state: &State,
+    fft_buf: &[Complex<f32>],
+    cmndf: &[f32],
+    current_frequency: f32,
+) where
+    B: Backend,
+{
+    let mut datasets = Vec::new();
+    let graph_data;
+    let x_bounds;
+    let y_bounds;
+    let cmndf_threshold;
+
+    match state.graph {
+        Graph::AmplitudeSpectrum => {
+            graph_data = fft_buf[1..FFT_SIZE / 2 + 1]
+                .iter()
+                .enumerate()
+                .map(|(index, &val)| {
+                    (
+                        f64::ln(((state.sample_rate / FFT_SIZE as f32) * index as f32).into()),
+                        20.0 * f64::log10(val.abs() as f64 / FFT_SIZE as f64),
+                    )
+                })
+                .collect::<Vec<_>>();
+
+            x_bounds = [
+                f64::ln((state.sample_rate / FFT_SIZE as f32).into()),
+                f64::ln(state.sample_rate as f64 / 2.0),
+            ];
+
+            y_bounds = [-140.0, 0.0];
+        }
+        Graph::CMNDF => {
+            graph_data = cmndf[1..]
+                .iter()
+                .enumerate()
+                .map(|(index, &val)| {
+                    (
+                        f64::ln((state.sample_rate / (index + 1) as f32).into()),
+                        val as f64,
+                    )
+                })
+                .collect::<Vec<_>>();
+
+            x_bounds = [
+                f64::ln(state.sample_rate as f64 / (cmndf.len() - 1) as f64),
+                f64::ln(state.sample_rate as f64 / 2.0),
+            ];
+
+            y_bounds = [0.0, 5.0];
+
+            cmndf_threshold = [
+                (x_bounds[0], CMNDF_THRESHOLD as f64),
+                (x_bounds[1], CMNDF_THRESHOLD as f64),
+            ];
+
+            datasets.push(
+                Dataset::default()
+                    .name("CMNDF threshold")
+                    .marker(symbols::Marker::Braille)
+                    .graph_type(GraphType::Line)
+                    .style(Style::default().fg(Color::Yellow))
+                    .data(&cmndf_threshold),
+            );
+        }
+    }
+
+    let target_freq;
+    if let Some(target_frequency) = state.target_frequency {
+        target_freq = Some([
+            (f64::ln(target_frequency.into()), y_bounds[0]),
+            (f64::ln(target_frequency.into()), y_bounds[1]),
+        ]);
```

Comment:
```suggestion
    if let Some(target_frequency) = state.target_frequency {
        let target_freq = [
            (f64::ln(target_frequency.into()), y_bounds[0]),
            (f64::ln(target_frequency.into()), y_bounds[1]),
        ];
```


It seems `target_freq` doesn't need to be wrapped in `Some`, you unwrap it immediately below in `datasets.push`